pkg/common/utils/utils.go

```diff
@@ -213,6 +213,66 @@ func GetApplicationIDFromPod(pod *v1.Pod) string {
 	return GenerateApplicationID(pod.Namespace, conf.GetSchedulerConf().GenerateUniqueAppIds, string(pod.UID))
 }
 
+func CheckAppIdInPod(pod *v1.Pod) error {
+	appIdLabelKeys := []string{
+		constants.CanonicalLabelApplicationID,
+		constants.SparkLabelAppID,
+		constants.LabelApplicationID,
+	}
+	appIdAnnotationKeys := []string{
+		constants.AnnotationApplicationID,
+	}
```

Comment:
These should be treated as immutable and always the same. The way we declare them now creates new objects and drops them on every call. We should pull them outside of the function so we have only one and document them .

Reply:
Thanks for the review. Moved appIdLabelKeys/appIdAnnotationKeys to constants.go in the new commits.